### PR TITLE
Ankit | stage | Prod - Invalid date error is coming when changing the entry date on ledger for imported transaction

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cw4245t

### CODE REVIEW

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 1]
```diff
@@ -59,6 +59,8 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
     private innerValue: any = '';
     /** This is used to show default date */
     public calendarDate: any = '';
+    /** This is used to show change date */
+    public inputChange : any ='';
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
    public inputChange: any ='';
```

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 2]
```diff
@@ -96,14 +98,27 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
...
     }
 
+    /**
+     * get current value on input
+     * 
+     * @param event 
+     */
+    public dateInputChange(event: Event): void {
+        if(event){
+        const inputElement = event.target as HTMLInputElement;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
            const inputElement = event.target as HTMLInputElement;
```

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 3]
```diff
@@ -96,14 +98,27 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
...
+        const inputValue = inputElement.value;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
            const inputValue = inputElement.value;
```

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 4]
```diff
@@ -96,14 +98,27 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
...
+        this.inputChange = inputValue;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
              this.inputChange = inputValue;
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: Formating

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 5]
```diff
@@ -96,14 +98,27 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
         this.destroyed$.complete();
     }
 
+    /**
+     * get current value on input
+     * 
+     * @param event 
+     */
+    public dateInputChange(event: Event): void {
+        if(event){
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
        if (event){
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts [hunk 6]
```diff
@@ -59,6 +59,8 @@ export class GiddhDatepickerComponent implements ControlValueAccessor, OnInit, O
     private innerValue: any = '';
     /** This is used to show default date */
     public calendarDate: any = '';
+    /** This is used to show change date */
+    public inputChange: any ='';
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]:
```suggestion
    public inputChange: any = '';
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

File: apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/theme/giddh-datepicker/giddh-datepicker.component.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

**Pull request closed without merging**